REGRESSION (iOS 18): 9 layout tests need to be re-written to expect new behavior.

charliewolfe opened the pull request:
#### 11ca8f29c7c6154703bcfee946a4009c67e90151
<pre>
REGRESSION (iOS 18): 9 layout tests need to be re-written to expect new behavior.
<a href="https://bugs.webkit.org/show_bug.cgi?id=277460">https://bugs.webkit.org/show_bug.cgi?id=277460</a>
<a href="https://rdar.apple.com/132946342">rdar://132946342</a>

Reviewed by Pascoe.

These tests should expect SameSite Lax cookies by default on iOS 18 and macOS Sequoia.

* LayoutTests/http/tests/cookies/resources/testharness-helpers.js:
* LayoutTests/http/tests/cookies/same-site/fetch-after-navigating-iframe-in-cross-origin-page-expected.txt:
* LayoutTests/http/tests/cookies/same-site/fetch-after-navigating-iframe-in-cross-origin-page.html:
* LayoutTests/http/tests/cookies/same-site/fetch-in-cross-origin-service-worker-expected.txt:
* LayoutTests/http/tests/cookies/same-site/fetch-in-cross-origin-service-worker.html:
* LayoutTests/http/tests/cookies/same-site/popup-cross-site-post.html:
* LayoutTests/http/tests/cookies/same-site/resources/fetch-after-navigating-iframe-in-cross-origin-page.py:
* LayoutTests/http/tests/cookies/same-site/resources/fetch-in-cross-origin-service-worker.html:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/cookie.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/cross-origin-embedder-policy/anonymous-iframe/cookie.tentative.https.window-expected.txt:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios-17/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/282548@main">https://commits.webkit.org/282548@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/953a63dcf350265ffb05c8568c5d3ceba5a4a558

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63380 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42736 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15977 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67401 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13988 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50424 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14268 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51052 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9668 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66449 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39678 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54887 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31733 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36359 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12237 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12860 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57898 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12564 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69097 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7327 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12165 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58361 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7359 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54958 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58601 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6109 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9593 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38557 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39636 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40748 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39379 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pascoej commented on LayoutTests/platform/ios/TestExpectations:
```diff
@@ -7488,16 +7488,14 @@ imported/w3c/web-platform-tests/mediacapture-insertable-streams/VideoTrackGenera
 imported/w3c/web-platform-tests/WebCryptoAPI/import_export/ec_importKey.https.any.worker.html [ Failure ]
 
-# webkit.org/b/277460 (REGRESSION (iOS 18): 9 layout tests need to be re-written to expect new behavior.)
 http/tests/cookies/same-site/fetch-after-navigating-iframe-in-cross-origin-page.html [ Failure ]
```
Can we re-add a comment explaining why the two are left here?

charliewolfe committed:
REGRESSION (iOS 18): 9 layout tests need to be re-written to expect new behavior.
https://bugs.webkit.org/show_bug.cgi?id=277460
rdar://132946342

Reviewed by Pascoe.

These tests should expect SameSite Lax cookies by default on iOS 18 and macOS Sequoia.

* LayoutTests/http/tests/cookies/resources/testharness-helpers.js:
* LayoutTests/http/tests/cookies/same-site/fetch-after-navigating-iframe-in-cross-origin-page-expected.txt:
* LayoutTests/http/tests/cookies/same-site/fetch-after-navigating-iframe-in-cross-origin-page.html:
* LayoutTests/http/tests/cookies/same-site/fetch-in-cross-origin-service-worker-expected.txt:
* LayoutTests/http/tests/cookies/same-site/fetch-in-cross-origin-service-worker.html:
* LayoutTests/http/tests/cookies/same-site/popup-cross-site-post.html:
* LayoutTests/http/tests/cookies/same-site/resources/fetch-after-navigating-iframe-in-cross-origin-page.py:
* LayoutTests/http/tests/cookies/same-site/resources/fetch-in-cross-origin-service-worker.html:
* LayoutTests/imported/w3c/web-platform-tests/html/anonymous-iframe/cookie.tentative.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/cross-origin-embedder-policy/anonymous-iframe/cookie.tentative.https.window-expected.txt:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios-17/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: https://commits.webkit.org/282548@main
